src/arkode/arkode_cli.c

```diff
@@ -0,0 +1,312 @@
+      {
+        arkProcessError(ark_mem, retval, __LINE__, __func__, __FILE__,
+                        "error setting command-line argument: %s",
+                        tworeal_pairs[j].key);
+        return retval;
+      }
+      if (arg_used) break;
+    }
+    if (arg_used) continue;
+
+    /* check all action command-line options */
+    for (j = 0; j < num_action_keys; j++)
+    {
+      retval = sunCheckAndSetActionArg(arkode_mem, &i, argv, offset,
+                                       action_pairs[j].key, action_pairs[j].set,
+                                       &arg_used);
+      if (retval != ARK_SUCCESS)
+      {
+        arkProcessError(ark_mem, retval, __LINE__, __func__, __FILE__,
+                        "error setting command-line argument: %s",
+                        action_pairs[j].key);
+        return retval;
+      }
+      if (arg_used) break;
+    }
+    if (arg_used) continue;
+
+    /*** handle all remaining command-line options ***/
+
+    if (strcmp(argv[i] + offset, "interpolant_type") == 0)
+    {
+      i++;
+      retval = ARK_ILL_INPUT;
+      if (strcmp(argv[i], "ARK_INTERP_HERMITE") == 0)
+      {
+        retval = ARKodeSetInterpolantType(arkode_mem, ARK_INTERP_HERMITE);
+      }
+      else if (strcmp(argv[i], "ARK_INTERP_LAGRANGE") == 0)
+      {
+        retval = ARKodeSetInterpolantType(arkode_mem, ARK_INTERP_LAGRANGE);
+      }
+      else if (strcmp(argv[i], "ARK_INTERP_NONE") == 0)
+      {
+        retval = ARKodeSetInterpolantType(arkode_mem, ARK_INTERP_NONE);
+      }
+      if (retval != ARK_SUCCESS)
+      {
+        arkProcessError(ark_mem, retval, __LINE__, __func__, __FILE__,
+                        "error setting command-line argument: %s %s",
+                        argv[i - 1], argv[i]);
+        return retval;
+      }
+      arg_used = SUNTRUE;
+      continue;
+    }
+
+    if (strcmp(argv[i] + offset, "accum_error_type") == 0)
+    {
+      i++;
+      retval = ARK_ILL_INPUT;
+      if (strcmp(argv[i], "ARK_ACCUMERROR_NONE") == 0)
+      {
+        retval = ARKodeSetAccumulatedErrorType(arkode_mem, ARK_ACCUMERROR_NONE);
+      }
+      else if (strcmp(argv[i], "ARK_ACCUMERROR_MAX") == 0)
+      {
+        retval = ARKodeSetAccumulatedErrorType(arkode_mem, ARK_ACCUMERROR_MAX);
+      }
+      else if (strcmp(argv[i], "ARK_ACCUMERROR_SUM") == 0)
+      {
+        retval = ARKodeSetAccumulatedErrorType(arkode_mem, ARK_ACCUMERROR_SUM);
+      }
+      else if (strcmp(argv[i], "ARK_ACCUMERROR_AVG") == 0)
+      {
+        retval = ARKodeSetAccumulatedErrorType(arkode_mem, ARK_ACCUMERROR_AVG);
+      }
+      if (retval != ARK_SUCCESS)
+      {
+        arkProcessError(ark_mem, retval, __LINE__, __func__, __FILE__,
+                        "error setting command-line argument: %s %s",
+                        argv[i - 1], argv[i]);
+        return retval;
+      }
+      arg_used = SUNTRUE;
+      continue;
+    }
+
+    if (strcmp(argv[i] + offset, "writeparameters") == 0)
```

Comment:
Is the convention snake case? Either way, I like that this option is included.
```suggestion
    if (strcmp(argv[i] + offset, "write_parameters") == 0)
```